cypress/support/pages/admin/pageAdministration.js

```diff
@@ -509,15 +496,30 @@ class Management {
 	}
 
 	enableStudentVisibilityForTeacher() {
-		cy.get(Management.#studentVisiblityToggle)
+		cy.get(Management.#studentVisibilityToggle)
 			.find("input")
 			.check({ force: true })
 			.wait(500);
-		cy.get(Management.#studentVisiblityToggle)
+		cy.get(Management.#studentVisibilityToggle)
 			.find("input")
 			.should("have.prop", "checked");
 	}
 
+	toggleStudentVisibilityForTeachersByAdmin(visibilityState) {
+		const desiredState = visibilityState === "enabled";
+		cy.get(Management.#studentVisibilityToggle)
+			.find("input")
+			.then(($input) => {
+				const isChecked = $input.prop("checked");
+				const isDisabled = $input.prop("disabled");
+				if (isDisabled) {
```

Comment:
no need if-else block. can be done like this
```suggestion
				if (isDisabled) {
					cy.log("Element is disabled, cannot toggle state");
					return;
				}
				if (isChecked === desiredState) {
					cy.log(`Element is already in the desired state, skipping click`);
					return;
				}
				cy.wrap($input).find("input").click({ force: true }).wait(500);
```